addons/account/models/account_move.py

```diff
@@ -1609,6 +1612,76 @@ def _compute_linked_attachment_id(self, attachment_field, binary_field):
     def _compute_incoterm_location(self):
         pass
 
+    @api.depends('partner_id', 'invoice_date', 'amount_total')
+    def _compute_abnormal_warning(self):
+        draft_invoices = self.filtered(lambda m: m.is_invoice() and m.state == 'draft' and m.amount_total)
+        other_moves = self - draft_invoices
+        other_moves.abnormal_amount_warning = False
+        other_moves.abnormal_date_warning = False
+        if draft_invoices:
+            draft_invoices.flush_recordset(['invoice_date', 'amount_total', 'partner_id', 'move_type', 'company_id'])
+            self.env.cr.execute("""
+                WITH previous_invoices AS (
+                      SELECT this.id,
+                             other.invoice_date,
+                             other.amount_total,
+                             LAG(other.invoice_date) OVER invoice - other.invoice_date AS date_diff
+                        FROM account_move this
+                        JOIN account_move other USING (partner_id, move_type, company_id, currency_id)
+                       WHERE other.state = 'posted'
+                         AND other.invoice_date <= this.invoice_date
+                         AND this.id = ANY(%s)
```

Comment:
`= ANY()` is checking that at least one of the elements in the SQL **array**(like) matches the left hand side.
It is very much expected to work 😄

Reply:
ha yes... 

note that we could use a subquery, using _search instead of filtering self, and having all wrapped in a single query. :thinking: but this query doesn't look like it needs more complexity, nor perf optimization... so it's probably fine as it

Reply:
But why would you do that, we already have the complete list of ids and we would need to inject that list of ids anyway in the `_search` :zany_face: